### PR TITLE
[RFR] Search as you type

### PR DESCRIPTION
**Pull request by ThieryMichel**
see #74 

capture du 2017-05-24 09 32 41

### CODE REVIEW

File: admin/src/Environment.js [hunk 1]
```diff
@@ -49,29 +55,33 @@ class Environment extends Component {
...
 
         return (
             <div style={styles.container}>
                 <Card>
-                    <CardText>
-                        {!edition && <RaisedButton label="Edit" primary onClick={toggleEdition} />}
-                        {edition && <RaisedButton label="Save" primary onClick={this.handleSaveClick} />}
-                        {edition && <RaisedButton label="Cancel" onClick={toggleEdition} />}
-                    </CardText>
+                    <CardActions style={styles.actions}>
```
Comment (djhi) [inline admin/src/Environment.js]: Why the change to `CardActions` ?

File: admin/src/Environment.js [hunk 2]
```diff
@@ -49,29 +55,33 @@ class Environment extends Component {
...
+                        <div>
```
Comment (djhi) [inline admin/src/Environment.js]: I think it would be better to move the edit button to the end and the search at the beginning because search is more important and will probably be used more often. Moreover, this full edition mode should be discouraged as potentially dangerous and we will introduce item edition soon (see #83)

But we should leave the Save and Cancel at the beginning in edition mode.

File: admin/src/Environment.js (hunk 1 above)
Comment (ThieryMichel) [inline admin/src/Environment.js]: Because it contained actions ?

Comment (djhi) [inline admin/src/Environment.js]: The CardActions component has no difference with the CardText component expect for padding

File: admin/src/Environment.js [hunk 3]
```diff
@@ -49,29 +55,39 @@ class Environment extends Component {
...
-                        {edition && <RaisedButton label="Cancel" onClick={toggleEdition} />}
+                        {edition &&
+                            <div>
+                                <RaisedButton label="Save" primary onClick={this.handleSaveClick} />
+                                <RaisedButton label="Cancel" onClick={toggleEdition} />
+                            </div>}
+                        {!edition &&
+                            <div style={styles.actions}>
+                                <TextField hintText="search" onChange={setSearch} />
+                                <div style={styles.search}>
```
Comment (ThieryMichel) [inline admin/src/Environment.js]: If we do not wrap the RaisedButton in a div, its display get bugged by the display flex on the container.

**Comment by djhi** [comment]
Can you update the screenshot ?

File: admin/src/Environment.js [hunk 4]
```diff
@@ -121,7 +121,7 @@ class Environment extends Component {
                             </List>
                         </CardText>}
                     {!loading &&
-                        config &&
+                        filteredConfig &&
```
Comment (djhi) [inline admin/src/Environment.js]: Not sure about that, it will only send the filtered config to the API right ?
